COM-841 Populate Radet report

lauravignoli opened the pull request.

lauravignoli committed:
COM-841 Populate Radet report

lauravignoli committed:
COM-841 Populate index testing tab

lauravignoli committed:
COM-841 Populate more fields

Fields on index testing and testing tabs.

lauravignoli committed:
COM-841 Added getObsCodedValueInSectionByNames function

tmvumbi2 committed:
COM-841 Populate fields treatment report
Populates some fields on the treatment report

lauravignoli committed:
COM-841 Add index calculation functions

Added getNumberOfHIVTestedContactsRelatedToIndex and getNumberOfHIVPosContactsRelatedToIndex.

tmvumbi2 committed:
Merge branch 'COM-841' of https://github.com/jembi/cameroon-bahmni-config into COM-841

tmvumbi2 committed:
COM-841 Populate fields pmtct report
Populates some fields on the pmtct report

lauravignoli committed:
COM-841 Populate more fields

For the index testing and testing tabs.

lauravignoli commented on metadata/reportssql/common_functions.sql:
```diff
@@ -90,4 +90,38 @@ END$$
 
 DELIMITER ; 
 
+-- calculateDurationInDays
```
@tmvumbi2 This is a function specific for drug duration, so I think it should go in the drug_functions file.

lauravignoli commented on openmrs/apps/reports/RADET/sql/pmtct_hei.sql:
```diff
@@ -1,42 +1,44 @@
+	"N/A" as "TB screening status at Last ARV Refill / Statut de dépistage de la tuberculose",
+	getViralLoadTestResult(p.patient_id) as "Current Viral Load/ Charge virale actuelle  (c/ml)",
+	getViralLoadTestDate(p.patient_id) as "Date of Current Viral Load / Date de la charge virale actuelle (dd-mmm-yyyy)",
+	"N/A" as "Viral Load Indication/ Indication de la charge virale",
```
@tmvumbi2 indication is the type of viral load test: Not documented, routine or targeted

lauravignoli commented on openmrs/apps/reports/RADET/sql/treatment.sql:
```diff
@@ -1,37 +1,38 @@
+	"N/A" as "Regimen Line at ART Start/ Ligne de régime à ART Start",
+	"N/A" as "Regimen at ART Start/ Régime chez ART Start", 
+	getPatientMostRecentProgramAttributeCodedValue(p.patient_id, "397b7bc7-13ca-4e4e-abc3-bf854904dce3", "en") as "Current Regimen Line/ Ligne de régime actuelle",
+	"N/A" as "Current ART Regimen/ Régime d'ART actuel",
```
@tmvumbi2 I think regimen is the list of active drugs

lauravignoli commented on openmrs/apps/reports/RADET/sql/treatment.sql:
```diff
@@ -1,37 +1,38 @@
+	"N/A" as "Date completed full course TPT/ Date de fin du cours complet TPT",
+	"N/A" as "Date of EAC 3 (dd/mm/yy)",
+	getPatientMostRecentProgramOutcome(p.patient_id, "en", "HIV_PROGRAM_KEY") as "Current ART Status (Active, LTFU, Dead, Transferred Out, Stopped)",
+	"N/A" as "Status of Missed appointment / Statut de rendez-vous manqué",
```
@tmvumbi2 Check the radet file, the statuses are defined there

tmvumbi2 committed:
COM-841 Move calculateDurationInDays to drug_functions.sql
This function is specific to drugs and should be in drug_function.sql

tmvumbi2 committed:
COM-841 Extend retrieveViralLoadTestDateAndResult to return the VL Indication
Adds the output viralLoadIndication to the function retrieveViralLoadTestDateAndResult: this approach avoids adding a new function to retrieve the vl indication and also won't too badly affect the existing function's performance (because the vl indication is retrieved from an existing 'select' query).

tmvumbi2 committed:
COM-841 Add Current ART Regimen to treatment report
Updates RADET/Treatment report: adds Current ART Regimen column

tmvumbi2 committed:
COM-841 Fix getObsCodedValueInSectionByNames: add LIMIT 1
The above function was causing a bug (ERROR 1172 (42000): Result consisted of more than one row) in RADET/testing report. The commit fixes the bug by adding LIMIT clause to the query in getObsCodedValueInSectionByNames

tmvumbi2 committed:
COM-841 Add column "status of missed appointment"
Adds a new function for the above column

tmvumbi2 committed:
COM-841 Include only patients that started ARV treat. in the treat. report
Updates the treatment report to include only patients for which:
- the treatment start date has been populated
- OR have picked up an ARV (note: this second condition has been requested by Emeka)

tmvumbi2 committed:
COM-841 Include only pregnant patients in the pmtct report
Include only patients for which the field "Date of First ANC Visit" (from the ANC initial form) has been populated

lauravignoli committed:
COM-841 Use risk group for KP status

tmvumbi2 committed:
COM-841 Add child's section of the pmtct report
Adds some more columns to the child's section of the pmtct report

tmvumbi2 committed:
COM-841 Add more columns to child's section of the pmtct report
Adds some more columns to the child's section of the pmtct report

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
+	getNumberOfHIVPosContactsRelatedToIndex(getFirstIndexID(p.patient_id)) as "No of contacts tested HIV positive/ Nombre de contacts testés séropositifs",
+	"N/A" as "APS"
+	FROM patient p
+		WHERE patientIsIndex(p.patient_id) OR patientIsContact(p.patient_id);
```
@lauravignoli should this report include only contacts, or both contact and index patients?

lauravignoli replied:
@tmvumbi2 I think both, otherwise you can't see index patients for which no contacts have been added yet?

lauravignoli replied:
You are right, we should include only contacts, otherwise the wrong info will be dispalyed.

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
-	0 as "Date of Contact elicitation from Index/ Date de sollicitation du contact à partir de l'index",
-	0 as "Date of birth (Date de naissance)",
+SELECT getPatientIdentifier(p.patient_id) as "Contacts (Contact nr/Child nr) ",
+	getPatientIndexTestingDateOffered(getFirstIndexID(p.patient_id)) as "Date of Contact elicitation from Index/ Date de sollicitation du contact à partir de l'index",
```
@lauravignoli isn't the accepted date closer to the value needed here? I am assuming that the date of contact elicitation from Index will always be equal or after the accepted date

lauravignoli replied:
@tmvumbi2 Cool will update.

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
-	0 as "APS";
\ No newline at end of file
+	getPatientGender(p.patient_id) as "Sex",
+	getPatientARTNumber(p.patient_id) as "Index Case Code (ART Code/ Testing Code*)",
```
@lauravignoli is this meant to be the ART number of the index related to the current contact? (i.e. `getPatientARTNumber(getFirstIndexID(p.patient_id))` )

lauravignoli replied:
@tmvumbi2 I think the index ARV number goes in line 21 getPatientARTNumber(getFirstIndexID(p.patient_id)) as "Client ART Unique ID/Patient Unique ID/ART TARV",

tmvumbi2 replied:
@lauravignoli  ah okay I see, thanks :+1:

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
\ No newline at end of file
+	getPatientGender(p.patient_id) as "Sex",
+	getPatientARTNumber(p.patient_id) as "Index Case Code (ART Code/ Testing Code*)",
+	getIndexType(p.patient_id) as "Type of Index Case/ type de cas index",
```
@lauravignoli is this meant to be the type of the index related to the current contact? (i.e. `getIndexType(getFirstIndexID(p.patient_id))` )

lauravignoli replied:
Yes, will update

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
+	getNumberOfHIVTestedContactsRelatedToIndex(getFirstIndexID(p.patient_id)) as "No of contacts tested for HIV/ Nombre de contacts testés pour le VIH",
+	getNumberOfHIVPosContactsRelatedToIndex(getFirstIndexID(p.patient_id)) as "No of contacts tested HIV positive/ Nombre de contacts testés séropositifs",
+	"N/A" as "APS"
+	FROM patient p
```
@lauravignoli we need to also check that the patient hasn't been voided

lauravignoli replied:
@tmvumbi2 As discussed, this is not necesssary.

tmvumbi2 commented on metadata/reportssql/index_program_functions.sql:
```diff
@@ -207,27 +273,129 @@ BEGIN
+
+    IF (daysBetweenHIVPosAndART > 0 AND daysBetweenHIVPosAndART <= 30) THEN
+        RETURN 'Index Case New HTS POS and initiated on treatment within a month';
+    ELSEIF (viralLoadResult > 1000) THEN
```
@lauravignoli the above condition looks different to the rest of the conditions (that are all related to daysBetweenHIVPosAndART). Isn't it possible to have a case where a patient is both 'Index Case virally unsuppressed client' and 'Index Case New HTS POS and initiated on treatment within a month'? Could we check the logic with Emeka and/or JP?

lauravignoli replied:
@tmvumbi2 I checked with Emeka before writing this:

![indexType](https://user-images.githubusercontent.com/1198827/83852037-7b1df700-a713-11ea-86e1-e5a9177e62df.png)

tmvumbi2 commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -1710,3 +1783,59 @@ BEGIN
+BEGIN
+    DECLARE artStartDate DATE;
+    SET artStartDate = getPatientARVStartDate(p_patientId);
+    RETURN (DATEDIFF(artStartDate, getHIVTestDate(p_patientId, "2000-01-01", artStartDate)));
```
@lauravignoli `getHIVTestDate` returns the hiv's test date irrespective of the result, for only positive result test date, we can use the function `getDateOfPositiveHIVResult`

lauravignoli replied:
@tmvumbi2 Good catch!!

tmvumbi2 commented on metadata/reportssql/testing_report_functions.sql:
```diff
@@ -1710,3 +1783,59 @@ BEGIN
+BEGIN
+    DECLARE enrollmentDate DATE;
+    SET enrollmentDate = getPatientDateOfEnrolmentInProgram(p_patientId, "HIV_PROGRAM_KEY");
+    RETURN (DATEDIFF(enrollmentDate, getHIVTestDate(p_patientId, "2000-01-01", enrollmentDate)));
```
@lauravignoli `getHIVTestDate` returns the hiv's test date irrespective of the result, for only positive result test date, we can use the function `getDateOfPositiveHIVResult`

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
+	"N/A" as "HTS Client ID (EID or HEI Number)",
+	getHIVTestDate(p.patient_id, "#startDate#", "#endDate#") as "Actual date of visit for HTS/ Date réelle de visite pour HTS (EID Sample Collected date/Échantillon EID Date de collecte)  (dd-MMM-yyyy)",
+	getHIVResult(p.patient_id, "#startDate#", "#endDate#") as "HTS Results/ Résultats HTS ",
+	getObsLocation(p.patient_id, "e7a002be-8afc-48b1-a81b-634e37f2961c") as "Location of testing/ emplacement des tests",
```
@lauravignoli is this the testing location of the index or contact?

lauravignoli replied:
I think because of where it is in the sheet, it should be the contact testing location. It seems that all that info is grouped together. But we can check with Emeka later today.

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
+	getHIVTestDate(p.patient_id, "#startDate#", "#endDate#") as "Actual date of visit for HTS/ Date réelle de visite pour HTS (EID Sample Collected date/Échantillon EID Date de collecte)  (dd-MMM-yyyy)",
+	getHIVResult(p.patient_id, "#startDate#", "#endDate#") as "HTS Results/ Résultats HTS ",
+	getObsLocation(p.patient_id, "e7a002be-8afc-48b1-a81b-634e37f2961c") as "Location of testing/ emplacement des tests",
+	getPatientDateOfEnrolmentInProgram(p.patient_id, "HIV_PROGRAM_KEY") as "Date Enrolled to HIV care / Date d'inscription aux soins du VIH (dd-MMM-yyyy)",
```
@lauravignoli is this the enrollment of the contact or index?

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/index_testing.sql:
```diff
@@ -1,33 +1,35 @@
+	getHIVResult(p.patient_id, "#startDate#", "#endDate#") as "HTS Results/ Résultats HTS ",
+	getObsLocation(p.patient_id, "e7a002be-8afc-48b1-a81b-634e37f2961c") as "Location of testing/ emplacement des tests",
+	getPatientDateOfEnrolmentInProgram(p.patient_id, "HIV_PROGRAM_KEY") as "Date Enrolled to HIV care / Date d'inscription aux soins du VIH (dd-MMM-yyyy)",
+	IF(getPatientDateOfEnrolmentInProgram(p.patient_id, "HIV_PROGRAM_KEY"), "Yes", "No") as "Linked to ART / Lié à ART",
```
@lauravignoli  we can use `patientHasEnrolledIntoHivProgram` here, it returns a yes/no answer

tmvumbi2 commented on openmrs/apps/reports/RADET/sql/testing.sql:
```diff
@@ -1,28 +1,29 @@
+	IF(patientIsIndex(p.patient_id),getNumberOfHIVTestedContactsRelatedToIndex(p.patient_id),0) as "No of contacts tested for HIV/ Nombre de contacts testés pour le VIH",
+	IF(patientIsIndex(p.patient_id),getNumberOfHIVPosContactsRelatedToIndex(p.patient_id),0) as "No of contacts tested HIV positive/ Nombre de contacts testés séropositifs",
+	"N/A" as "APS"
+	FROM patient p WHERE getHIVResult(p.patient_id, "2000-01-01", "2050-01-01") IS NOT NULL;
```
@lauravignoli we need to also check that the patient hasn't been voided

lauravignoli replied:
Not necessary :)

lauravignoli committed:
COM-841 Add VL indication to treatment tab

tmvumbi2 committed:
COM-841 Include only non-voided patients into the treatment report
Excludes all the patients that have been soft-deleted from the radet/treatment report

lauravignoli committed:
COM-841 Updates to testing and index testing tabs
